NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -1,45 +1,129 @@
+  return GradedAxes.gradedrange(LabelledNumbers.labelled.(degen, C.(labels)))
 end
 
-function trivial(category_type::Type{<:AbstractCategory})
-  return error("`trivial` not defined for type $(category_type).")
+# abelian case: return Category
+function fusion_rule(::AbelianGroup, c1::C, c2::C) where {C<:AbstractCategory}
+  return C(label_fusion_rule(C, category_label(c1), category_label(c2)))
 end
 
-istrivial(c::AbstractCategory) = (c == trivial(typeof(c)))
+function fusion_rule(
+  ::EmptyCategory, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  return LabelledNumbers.labelled(l1 * l2, sector())
+end
+
+function label_fusion_rule(category_type::Type{<:AbstractCategory}, ::Any, ::Any)
+  return error("`label_fusion_rule` not defined for type $(category_type).")
+end
 
-function dual(category_type::Type{<:AbstractCategory})
-  return error("`dual` not defined for type $(category_type).")
+# ================================  GradedAxes interface  ==================================
+# tensor_product interface
+function GradedAxes.fuse_blocklengths(
+  l1::LabelledNumbers.LabelledInteger{<:Integer,<:Sectors.AbstractCategory},
+  l2::LabelledNumbers.LabelledInteger{<:Integer,<:Sectors.AbstractCategory},
+)
+  return GradedAxes.fuse_blocklengths(
+    combine_styles(SymmetryStyle(l1), SymmetryStyle(l2)), l1, l2
+  )
 end
 
-Base.isless(c1::AbstractCategory, c2::AbstractCategory) = isless(label(c1), label(c2))
+function GradedAxes.fuse_blocklengths(
+  ::SymmetryStyle, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  fused = LabelledNumbers.label(l1) ⊗ LabelledNumbers.label(l2)
+  v =
+    LabelledNumbers.labelled.(
+      l1 * l2 .* BlockArrays.blocklengths(fused), GradedAxes.blocklabels(fused)
+    )
+  return GradedAxes.gradedrange(v)
+end
+
+function GradedAxes.fuse_blocklengths(
+  ::AbelianGroup, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  fused = LabelledNumbers.label(l1) ⊗ LabelledNumbers.label(l2)
+  return LabelledNumbers.labelled(l1 * l2, fused)
+end
+
+function GradedAxes.fuse_blocklengths(
+  ::EmptyCategory, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  return LabelledNumbers.labelled(l1 * l2, sector())
+end
+
+# cast to range
+to_graded_axis(c::AbstractCategory) = to_graded_axis(LabelledNumbers.labelled(1, c))
+to_graded_axis(l::LabelledNumbers.LabelledInteger) = GradedAxes.gradedrange([l])
+to_graded_axis(g::AbstractUnitRange) = g
+
+# allow to fuse a category with a GradedUnitRange
+function GradedAxes.tensor_product(c::AbstractCategory, g::AbstractUnitRange)
+  return GradedAxes.tensor_product(to_graded_axis(c), g)
+end
+
+function GradedAxes.tensor_product(g::AbstractUnitRange, c::AbstractCategory)
+  return GradedAxes.tensor_product(c, g)
+end
+
+function GradedAxes.tensor_product(c1::AbstractCategory, c2::AbstractCategory)
+  return to_graded_axis(fusion_rule(c1, c2))
```

Comment:
```suggestion
  return GradedAxes.tensor_product(to_graded_axis(c1), to_graded_axis(c2))
```
to follow the code patterns of the other definitions and not be too presumptuous about how `GradedAxes.tensor_product` is implemented.

Reply:
These are not strictly equivalent. `fusion_rule(::NonAbelian,::NonAbelian)` already returns a `GradedOneTo` and `to_graded_axis` has no effect.  `fusion_rule(::Abelian,::Abelian)` returns an `AbstractCategory` and `to_graded_axis` ensures that it is cast to a `GradedOneTo`. Calling `to_graded_axis` before would make the method dependent on `tensor_product(::GradedOneTo,::GradedOneTo)` implementation details (and be less efficient as it would involve many other operations).

Reply:
I think I understand your point, however I am confused by why you say:

> These are not strictly equivalent.

Your original code and my suggestion must output the same thing, right? In other words, by "equivalent" do you mean equivalent implementation, or equivalent output?

Reply:
It is not equivalent implementation. The output will be the same indeed.

Reply:
Thanks for the clarification, I agree my suggestion is a different implementation and may even be a bit slower, I'm generally erring on the side of keeping code conceptually simpler at the expense of being a bit slower at this point, and then optimizing as needed later, but this is a case I could go either way about since your original code is pretty simple anyway. So I'm fine keeping it as is, though please do keep in mind that tradeoff in other code.